generator/src/Pos/Generator/Block/Logic.hs

```diff
@@ -137,22 +145,57 @@ genBlock pm eos = do
                     -- When we know the secret key we can proceed to the actual creation.
                     Just <$> usingPrimaryKey leaderSK
                              (lift $ genMainBlock slot (swap <$> transCert))
-  where
+    where
     genMainBlock ::
         SlotId ->
         ProxySKBlockInfo ->
-        BlockGenMode (MempoolExt m) m Blund
+        BlockGenMode (MempoolExt m) m Block
     genMainBlock slot proxySkInfo =
         createMainBlockInternal pm slot proxySkInfo >>= \case
             Left err -> throwM (BGFailedToCreate err)
-            Right mainBlock -> verifyAndApply $ Right mainBlock
-    verifyAndApply :: Block -> BlockGenMode (MempoolExt m) m Blund
-    verifyAndApply block =
-        verifyBlocksPrefix pm (one block) >>= \case
+            Right mainBlock -> return $ Right mainBlock
+
+-- | Generate a valid 'Block' for the given epoch or slot (genesis block
+-- in the former case and main block the latter case) and apply it.
+genBlock ::
+       forall g ctx m.
+       ( RandomGen g
+       , MonadBlockGen ctx m
+       , Default (MempoolExt m)
+       , MonadTxpLocal (BlockGenMode (MempoolExt m) m)
+       , HasTxpConfiguration
```

Comment:
Is it feasible to remove this and take the `TxpConfiguration` as a parameter? Or would that cascade into a whole lot of refactoring?